### DISCUSSION TITLE
SFCC - OCAPI

### DISCUSSION THREAD

**Discussion by Cororo** [SalesforceCommerceCloud/sfcc-ci#431] [Team Posts]
Hi all,

I am using OCAPI to create/update products. It works well except for some system fields such as "taxClassID" or "pageDescription".

When doing the OCAPI requests it is giving this error: 

```
{
    "_v": "19.5",
    "fault": {
        "arguments": {
            "document": "product",
            "property": "pageDescription"
        },
        "type": "UnknownPropertyException",
        "message": "An error occurred while decoding the request. There's an unknown property 'pageDescription' in document 'product'."
    }
}
```

Is there a reason why those fields can't be set via OCAPI? Or am I doing something wrong?

Thanks

**Comment by bmoraux** [comment]
From the documentation the field is not "pageDescription" but "page_description", and "taxClassID" does not exist in product Update.

https://documentation.b2c.commercecloud.salesforce.com/DOC1/topic/com.demandware.dochelp/OCAPI/19.5/data/Documents/Product.html

Regards,

**Comment by uditshastri23** [comment]
Hi,  You may refer to this link for OCAPI product document properties:
https://documentation.b2c.commercecloud.salesforce.com/DOC1/topic/com.demandware.dochelp/OCAPI/19.5/data/Documents/Product.html?resultof=%22%64%6f%63%75%6d%65%6e%74%22%20%22%70%72%6f%64%75%63%74%22%20
.

"page_description" is the right property name for this.  Tax Class ID does
not exist in the property list , but you can modify a custom attribute
using OCAPI which can be set to class ID value.



On Tue, Jul 9, 2019 at 2:41 PM Cororo <notifications@github.com> wrote:

> Hi all,
>
> I am using OCAPI to create/update products. It works well except for some
> system fields such as "taxClassID" or "pageDescription".
>
> When doing the OCAPI requests it is giving this error:
>
> {
>     "_v": "19.5",
>     "fault": {
>         "arguments": {
>             "document": "product",
>             "property": "pageDescription"
>         },
>         "type": "UnknownPropertyException",
>         "message": "An error occurred while decoding the request. There's an unknown property 'pageDescription' in document 'product'."
>     }
> }
>
> Is there a reason why those fields can't be set via OCAPI? Or am I doing
> something wrong?
>
> Thanks
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/SalesforceCommerceCloud/teams/community/discussions/82?email_source=notifications&email_token=ALXSBK7FD6HNZCZRBX6KU73P6RI2FA5CNFSM4H7DJ4LKYY3PNVWWK3TUL52HS4DFVZCGS43DOVZXG2LPNZIG643UVJRW63LNMVXHIX3JMTHAAAI26Q>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ALXSBKZYJ7EEWYY4Q3RB3ALP6RI2FANCNFSM4H7DJ4LA>
> .
>


-- 

*Udit Shastri*

*Web Developer*

———

EMAIL

u.shastri@astoundcommerce.com

DIRECT

+91 9971024824

Be Astounding.

www.astoundcommerce.com
<http://astoundcommerce.com/?utm_source=signature&utm_medium=email&utm_campaign=GL-mrkt>

[image: linked.png]
<http://t.sidekickopen17.com/e1t/c/5/f18dQhb0S7lC8dDMPbW2n0x6l2B9nMJW7t5XZs4XyQDxW5wvLN42BpCD8W2zq3QW56dzCyf85PTwq02?t=https%3A%2F%2Fwww.linkedin.com%2Fcompany%2F2403308&si=5196493772554240&pi=c5db25f7-674e-4aa4-c4a2-7c81e0bb8946>

[image: twitter.png]
<http://t.sidekickopen17.com/e1t/c/5/f18dQhb0S7lC8dDMPbW2n0x6l2B9nMJW7t5XZs4XyQDxW5wvLN42BpCD8W2zq3QW56dzCyf85PTwq02?t=https%3A%2F%2Ftwitter.com%2FAstoundCommerce&si=5196493772554240&pi=c5db25f7-674e-4aa4-c4a2-7c81e0bb8946>

[image: logo.gif]

-- 
This e-mail, including any attachments, contains confidential information 
of Astound Commerce or its affiliates (collectively "Astound"), and/or its 
clients. It may be read, copied and used only by the intended recipient. 
Any use by a person other than its intended recipient, or by the recipient 
but for purposes other than the intended purpose, is strictly prohibited. 
If you received this e-mail in error, please contact the sender and then 
destroy this e-mail. Opinions, conclusions and other information in this 
message that do not relate to the official business of Astound shall be 
understood to be neither given nor endorsed by Astound.

**Comment by Cororo** [comment]
Thanks, I was taking the ID that was defined in the system object type "pageDescription". That works now.

@uditshastri23 I don't understand what you said about taxClassID. In my system object type it is said that it is an system attribute. So is it missing from the product document? Because if I do the OCAPI request with "c_taxClassID" there is no error, but the field is not updated because it is not a custom field.

**Comment by Cororo** [comment]
I am having the same issue with "searchPlacement". I understand that it doesn't work because it is not in the Produt document.
But how can we update those fields via OCAPI?

Those fields are system attribute 
![image](https://user-images.githubusercontent.com/3299091/60886634-cbb27900-a252-11e9-9eee-cd322d18772a.png)

Is there a work around/special way to do it? Or is it just not possible?

Thanks
